src/EnergyPlus/fenv_missing.h

```diff
@@ -0,0 +1,147 @@
+    // assert((FE_FLUSHTOZERO << 8) != __fpcr_flush_to_zero);
+    // assert((FE_FLUSHTOZERO << 8) == __fpcr_trap_denormal);
+    static constexpr unsigned int FPCR_EXCEPT_SHIFT = 8;
+
+    // These will be the exception flags we use for exception values normalized
+    // from both status word and control word.
+    // We add EX_ prefix to the names since macOS <math.h> defines OVERFLOW and
+    // UNDERFLOW macros.
+    static constexpr uint32_t EX_INVALID = 0x1;
+    static constexpr uint32_t EX_DIVBYZERO = 0x2;
+    static constexpr uint32_t EX_OVERFLOW = 0x4;
+    static constexpr uint32_t EX_UNDERFLOW = 0x8;
+    static constexpr uint32_t EX_INEXACT = 0x10;
+    // __APPLE__ ARM64 has an extra flag that is raised when a denormal is flushed
+    // to zero.
+    static constexpr uint32_t EX_FLUSHTOZERO = 0x20;
+
+    inline int feenableexcept(unsigned int excepts)
+    {
+      static fenv_t fenv;
+      if (std::fegetenv(&fenv) != 0) {
+        return -1;
+      }
+      const unsigned long long old_fpcr = fenv.__fpcr;
+      const unsigned int old_excepts = (old_fpcr >> FPCR_EXCEPT_SHIFT) & unsigned(FE_ALL_EXCEPT);
+
+      // Check the bits passed are valid, and bit shift them
+      const unsigned int new_excepts = excepts & unsigned(FE_ALL_EXCEPT);
+      const unsigned long long  new_fpcr = new_excepts << FPCR_EXCEPT_SHIFT;
+
+      // Set the new bits
+      fenv.__fpcr = fenv.__fpcr | new_fpcr;
+
+      return (std::fesetenv(&fenv) != 0) ? -1 : static_cast<int>(old_excepts);
+    }
+
+    inline int fedisableexcept(unsigned int excepts)
+    {
+      static fenv_t fenv;
+      if (std::fegetenv(&fenv) != 0) {
+        return -1;
+      }
+
+      const unsigned long long old_fpcr = fenv.__fpcr;
+      const unsigned int old_excepts = ((old_fpcr >> FPCR_EXCEPT_SHIFT) &  unsigned(FE_ALL_EXCEPT));
+
+      // Check the bits passed are valid, and bit shift them
+      const unsigned int new_excepts = excepts &  unsigned(FE_ALL_EXCEPT);
+      const unsigned long long  new_fpcr = new_excepts << FPCR_EXCEPT_SHIFT;
+
+      // Clear the bits
+      fenv.__fpcr &= ~new_fpcr;
+
+      return std::fesetenv(&fenv) ? -1 : static_cast<int>(old_excepts);
+    }
+
+#    endif
+
+#  else
+  inline int feenableexcept(unsigned int excepts)
+  {
+  #pragma STDC FENV_ACCESS ON
+    fexcept_t flags;
+    /* Save current exception flags. */
+    fegetexceptflag(&flags, FE_ALL_EXCEPT);
+
+    feclearexcept(FE_ALL_EXCEPT); /* clear all fp exception conditions */
+    return fesetexceptflag(&flags, excepts) != 0 ? -1 : flags; /* set new flags */
+  }
+
+  inline int fedisableexcept(unsigned int excepts)
+  {
+  #pragma STDC FENV_ACCESS ON
+    fexcept_t flags;
+    /* Save current exception flags. */
+    fegetexceptflag(&flags, FE_ALL_EXCEPT);
+
+    feclearexcept(FE_ALL_EXCEPT); /* clear all fp exception conditions */
+    return fesetexceptflag(&flags, ~excepts) != 0 ? -1 : flags; /* set new flags */
+  }
+
+#  endif
+#endif
+
+// clang-format on
```

Comment:
This file is for mac. Implements the feeenableexcept. I pwned the x86 one, spent a bunch of time making a clean one for arm64 (M1 macs)